powershell.go

```diff
@@ -3,21 +3,28 @@ package winrm
 import (
 	"encoding/base64"
 	"fmt"
+	"unicode/utf16"
 )
 
 // Powershell wraps a PowerShell script
 // and prepares it for execution by the winrm client
 func Powershell(psCmd string) string {
-	// 2 byte chars to make PowerShell happy
-	wideCmd := ""
-	for _, b := range []byte(psCmd) {
-		wideCmd += string(b) + "\x00"
-	}
+	// Encode the command string as a Windows wide-character string (UTF-16LE).
+	wideCmd := encodeUtf16Le(psCmd)
 
 	// Base64 encode the command
-	input := []uint8(wideCmd)
-	encodedCmd := base64.StdEncoding.EncodeToString(input)
+	encodedCmd := base64.StdEncoding.EncodeToString(wideCmd)
 
 	// Create the powershell.exe command line to execute the script
 	return fmt.Sprintf("powershell.exe -EncodedCommand %s", encodedCmd)
 }
+
+func encodeUtf16Le(s string) []byte {
+	d := utf16.Encode([]rune(s))
+	b := make([]byte, len(d)*2)
+	for i, r := range d {
+		b[i*2] = byte(r)
```

Comment:
I think to be on the safe side that it would be better to use:
```
binary.LittleEndian.PutUint16(b[i*2:], r)
```

Reply:
Why do you think its unsafe?

Reply:
Maybe safe wasn't the correct word. It's definitely easier to use the standard library way of doing things and possibly correct on all platforms.